packages/react/src/ActionList/Item.tsx

```diff
@@ -24,6 +24,14 @@ import VisuallyHidden from '../_VisuallyHidden'
 
 const LiBox = styled.li<SxProp>(sx)
 
+const ButtonItemContainer = React.forwardRef(({as: Component = 'button', children, styles, ...props}, forwardedRef) => {
+  return (
+    <Box as={Component as React.ElementType} ref={forwardedRef} sx={styles} {...props}>
```

Comment:
I saw we were already handling this on [line #329](https://github.com/primer/react/blob/63816188a218645de10fe9fcb7773356fb9724c5/packages/react/src/ActionList/Item.tsx#L329C21-L329C67), so now we pass the same logic to `ButtonItemContainer`, but under the styles prop instead of handling it inside of the component.